src/librustc_traits/dropck_outlives.rs

```diff
@@ -241,6 +247,10 @@ fn dtorck_constraint_for_ty<'tcx>(
             // derived from lifetimes attached to the upvars and resume
             // argument, and we *do* incorporate those here.
 
+            if !substs.as_generator().is_valid() {
```

Comment:
Add comment here 
```
            // By the time this code runs, all type variables ought to		            // By the time this code runs, all type variables ought to
            // be fully resolved.
```

Same as the `ty::Infer` case

Reply:
hmmm well im already deleting my folder so i can  use ssh
so that it a later issue